pyccel/ast/numpyext.py

```diff
@@ -1202,27 +1202,15 @@ def __init__(self, x):
 
 class NumpyMod(NumpyUfuncBinary):
     def __init__(self, x1, x2):
-        self._shape     = x1.shape
-        self._rank      = x1.rank
-        self._dtype     = x1.dtype
-        self._precision = x1.precision
-        self._order     = x1.order
+        self.copy(x1)
```

Comment:
We need to be careful here:
```
5 % np.arange(1, 5)
```
should return `array([0, 1, 2, 1])`, so copying the attributes of `5` will give the wrong result.

Reply:
ah yes you are right, what I did was wrong

Reply:
done